### ISSUE TITLE
ltn12 should not export global table

### ISSUE THREAD

**Issue by johnd0e** [lunarmodules/luasocket#443]
https://github.com/lunarmodules/luasocket/blob/1fad1626900a128be724cba9e9c19a6b2fe2bf6b/src/ltn12.lua#L16-L19

`module` function exists in 5.1 (in 5.2 manual it is referenced as deprecated).
Still nobody nowadays expect that module exports globals.
Nor other (sub)modules of `luasocket` do that.

**Comment by alerque** [comment]
I agree this should be removed, but doing so will probably warrant a breaking release semver bump because it will not be a smooth upgrade path for anybody that *was* relying on this quirk.

**Comment by catwell** [comment]
A long time ago almost all Lua modules used to export a global. ltn12 is a module from that era and used in a lot of places. So yes I can guarantee you changing this will break code, it should be done in a major revision.
